linuxkit/linuxkit
# [WIP] Add TPM module builder options to kernel

**- What I did**
Added all of the in-tree tpm options to the kernel config

**- How I did it**
1. Run the kernel build container
2. Stop it to get to the build image hash
3. Run the build hash with the `kernel/` dir mounted in
4. Run `make menuconfig`
5. Load in the existing `.config`
6. Add tpm options all is `M` (as in modules)
7. Save config to `.config`
8. Copy config to bind mounted dir
9. Exit container
10. Compare the config with the original config, fix any errors (mainly `HYPERV_SOCK` missing)
11. Copy over to appropriate `kernel.config-*` and build

**- How to verify it**
1. Build a kernel. 
2. Run on a virtual machine (e.g. hyperkit) without tpm, see it works, no impact, modules not loaded
3. Run on a machine with tpm (virtual or physical):
    * see modules are loaded
    * check that `/dev/tpm0` is created
    * check that `/sys/class/tpm/...` is properly populated with device info
    * check that `../pcrs` has actual information
    * check that `tcsd` can start and can get `tpm_version`


**- Description for the changelog**

Add support via kernel modules for in-tree tpm drivers

**- A picture of a cute animal (not mandatory but encouraged)**
![mongoose](https://farm8.static.flickr.com/7447/14136736111_a0528d38d4.jpg)

As discussed with @justincormack and @rn.

**Note:** This is still WIP. The current PR only includes changes for 4.9.x series kernel. Working on 4.11.x
